File: ems-frontend/src/services/leaveService.js
import API from "../api/axiosConfig";

export const applyLeave = (data) =>
    API.post("/leaves", data);

export const getMyLeaves = () =>
    API.get("/leaves/me");

export const getAllLeaves = () =>
    API.get("/leaves");

export const approveLeave = (id) =>
    API.put(`/leaves/${id}/approve`);

export const rejectLeave = (id) =>
    API.put(`/leaves/${id}/reject`);

export const getLeaveStatistics = () =>
    API.get("/leaves/statistics");

export const getLeavesByStatus = (status) =>
    API.get(`/leaves/status?status=${status}`);

File: EmployeeManagementSystem/src/main/java/org/mahesh/ems/service/LeaveService.java
package org.mahesh.ems.service;

import org.mahesh.ems.dto.LeaveRequestDto;
import org.mahesh.ems.dto.LeaveResponse;
import org.mahesh.ems.dto.LeaveStatisticsResponse;
import org.mahesh.ems.entity.LeaveStatus;

import java.util.List;

public interface LeaveService {

    LeaveResponse applyLeave(String email, LeaveRequestDto request);

    List<LeaveResponse> getMyLeaves(String email);

    List<LeaveResponse> getAllLeaves();

    LeaveResponse approveLeave(Long id);

    LeaveResponse rejectLeave(Long id);

    LeaveStatisticsResponse getLeaveStatistics();
    List<LeaveResponse> getLeavesByStatus(LeaveStatus status);
}